strongbox-user-management/src/test/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapContextLoaderTest.java

```diff
@@ -0,0 +1,159 @@
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = { UsersConfig.class,
+                                  AuthenticationConfig.class })
+public class LdapContextLoaderTest
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(LdapContextLoaderTest.class);
+
+
+    GenericXmlApplicationContext appCtx;
+
+    @Inject
+    private ApplicationContext parentApplicationContext;
+
+    @Before
+    public void initAppContext()
+            throws IOException
+    {
+        if (appCtx == null)
+        {
+            appCtx = new GenericXmlApplicationContext();
+
+            appCtx.setParent(parentApplicationContext);
+            appCtx.load(getAuthenticationConfigurationResource());
+            appCtx.refresh();
+        }
+    }
+
+    @After
+    public void closeAppContext()
+    {
+        if (appCtx != null)
+        {
+            appCtx.close();
+            appCtx = null;
+        }
+    }
+
+
+    private void displayBeanNames()
+    {
+        if (appCtx != null)
+        {
+            Stream.of(appCtx.getBeanDefinitionNames())
+                  .forEach(logger::debug);
+        }
+    }
+
+    @Test
+    public void embeddedLdapServerCreationContainsExpectedContextSourceAndData()
+    {
+
+        DefaultSpringSecurityContextSource contextSource = (DefaultSpringSecurityContextSource) appCtx.getBean(
+                "contextSource");
+
+        LdapTemplate template = new LdapTemplate(contextSource);
+        Object ldapObject = template.lookup("uid=przemyslaw.fusik,ou=Users");
+
+        Assert.assertThat(ldapObject, CoreMatchers.notNullValue());
+        Assert.assertThat(ldapObject, CoreMatchers.instanceOf(DirContextAdapter.class));
+        Assert.assertThat(((DirContextAdapter) ldapObject).getDn()
+                                                          .toString(),
+                          CoreMatchers.equalTo("uid=przemyslaw.fusik,ou=Users"));
+        Assert.assertThat(((DirContextAdapter) ldapObject).getNameInNamespace(),
+                          CoreMatchers.equalTo("uid=przemyslaw.fusik,ou=Users,dc=carlspring,dc=com"));
+    }
+
+    @Test
+    public void embeddedLdapServerRegistersExpectedAuthenticationProvider()
+    {
+
+        LdapAuthenticationProvider LdapAuthenticationProvider = (org.springframework.security.ldap.authentication.LdapAuthenticationProvider) appCtx.getBean(
+                "ldapAuthenticationProvider");
+        Authentication authentication = LdapAuthenticationProvider.authenticate(
+                new UsernamePasswordAuthenticationToken("przemyslaw.fusik", "password"));
+
+        Assert.assertThat(authentication, CoreMatchers.notNullValue());
+        Assert.assertThat(authentication, CoreMatchers.instanceOf(UsernamePasswordAuthenticationToken.class));
+        Assert.assertThat(authentication.getPrincipal(), CoreMatchers.instanceOf(LdapUserDetailsImpl.class));
+        Assert.assertThat(((LdapUserDetailsImpl) authentication.getPrincipal()).getDn(),
```

Comment:
Cosmetic: Wouldn't it be better to just define a variable here for the principal and just re-use it, instead of all the casting?